components/script/dom/document.rs

```diff
@@ -2772,7 +2772,14 @@ impl DocumentMethods for Document {
         if self.is_html_document {
             local_name.make_ascii_lowercase();
         }
-        let name = QualName::new(ns!(html), LocalName::from(local_name));
+
+        let ns = if self.is_html_document || self.content_type == DOMString::from("application/xhtml+xml") {
```

Comment:
Don't create a `DOMString` just for the equality check, the type `DOMString` implements `PartialEq<&str>`.

```rust
self.content_type == "application/xhtml+xml"
```